python/tank/util/defaults_manager.py

```diff
@@ -60,7 +60,7 @@ def get_http_proxy(self):
                   None if not necessary.
         """
         from . import shotgun
-        return shotgun.get_associated_sg_config_data().get("http_proxy")
+        return shotgun.get_associated_sg_config_data().get("http_proxy") or super(CoreDefaultsManager, self).get_http_proxy()
```

Comment:
After offline discussions with @robblau and @manneohrstrom , it was decided that the implementation wouldn't change as we couldn't come up with a strong workflow that would benefit from looking up the values the other way around.